plugins/inputs/cpu/cpu.go

```diff
@@ -113,6 +125,25 @@ func (c *CPUStats) Gather(acc telegraf.Accumulator) error {
 	return err
 }
 
+func (c *CPUStats) Init() error {
+	if c.CoreTags {
+		cpuInfo, err := cpuUtil.Info()
+		if err == nil {
+			c.coreID = cpuInfo[0].CoreID != ""
+			c.physicalID = cpuInfo[0].PhysicalID != ""
+
+			c.cpuInfo = make(map[string]cpuUtil.InfoStat)
+			for _, ci := range cpuInfo {
+				c.cpuInfo[fmt.Sprintf("cpu%d", ci.CPU)] = ci
+			}
+		} else {
+			c.Log.Warnf("Failed to gather info about CPUs: %s", err)
+		}
```

Comment:
How about being more strict here?
```suggestion
		if err != nil {
			return fmt.Errorf("failed to gather info about CPUs: %s", err)
		}
		c.coreID = cpuInfo[0].CoreID != ""
		c.physicalID = cpuInfo[0].PhysicalID != ""

		c.cpuInfo = make(map[string]cpuUtil.InfoStat)
		for _, ci := range cpuInfo {
			c.cpuInfo[fmt.Sprintf("cpu%d", ci.CPU)] = ci
		}
```

Reply:
I wanted to avoid stopping telegraf with an error when it in general does not block collecting cpu metrics, only the tags will be missing.

Reply:
Fair enough.